colossalai/booster/plugin/torch_fsdp_plugin.py

```diff
@@ -0,0 +1,202 @@
+        self.save_checkpoint(optim_state_dict, checkpoint)
+
+
+class TorchFSDPModel(ModelWrapper):
+
+    def __init__(self, module: nn.Module, *args, **kwargs) -> None:
+        super().__init__(module)
+        self.module = FSDP(module, *args, **kwargs)
+
+    def unwrap(self):
+        return self.module.module
+
+
+class TorchFSDPlugin(DPPluginBase):
+    """
+    Plugin for PyTorch FSDP.
+
+    Example:
+        >>> from colossalai.booster import Booster
+        >>> from colossalai.booster.plugin import TorchFSDPPlugin
+        >>>
+        >>> model, train_dataset, optimizer, criterion = ...
+        >>> plugin = TorchFSDPPlugin()
+
+        >>> train_dataloader = plugin.prepare_train_dataloader(train_dataset, batch_size=8)
+        >>> booster = Booster(plugin=plugin)
+        >>> model, optimizer, train_dataloader, criterion = booster.boost(model, optimizer, train_dataloader, criterion)
+
+    Args:
+        See https://pytorch.org/docs/stable/fsdp.html for details.
+    """
+
+    def __init__(self,
+                 sharding_strategy: Optional[ShardingStrategy] = None,
+                 cpu_offload: Optional[CPUOffload] = None,
+                 auto_wrap_policy: Optional[Union[Callable, _FSDPPolicy]] = None,
+                 backward_prefetch: Optional[BackwardPrefetch] = BackwardPrefetch.BACKWARD_PRE,
+                 mixed_precision: Optional[MixedPrecision] = None,
+                 ignored_modules: Optional[Iterable[torch.nn.Module]] = None,
+                 param_init_fn: Optional[Callable[[nn.Module], None]] = None,
+                 device_id: Optional[Union[int, torch.device]] = None,
+                 sync_module_states: bool = False,
+                 forward_prefetch: bool = False,
+                 limit_all_gathers: bool = False,
+                 use_orig_params: bool = False,
+                 ignored_parameters: Optional[Iterable[torch.nn.Parameter]] = None,
+    ):
+        super().__init__()
+        self.fsdp_kwargs = dict(sharding_strategy=sharding_strategy,
+                               cpu_offload=cpu_offload,
+                               auto_wrap_policy=auto_wrap_policy,
+                               backward_prefetch=backward_prefetch,
+                               mixed_precision=mixed_precision,
+                               ignored_modules=ignored_modules,
+                               param_init_fn=param_init_fn,
+                               device_id=device_id,
+                               sync_module_states=sync_module_states,
+                               forward_prefetch=forward_prefetch,
+                               limit_all_gathers=limit_all_gathers,
+                               use_orig_params=use_orig_params,
+                               ignored_parameters=ignored_parameters)
+
+    def support_no_sync(self) -> bool:
+        return True
+
+    def control_precision(self) -> bool:
+        return False
```

Comment:
I notice that there is a `mixed_precision` argument of this class. Are you sure that this class does not control precision?